Node.js v15 compatibility?

Hey,

I'm getting an interesting error while checking the types using node.js v15 (runtime, not `@types/node`).

To run the check in node v14 you can do this:

```
git clone https://github.com/signicode/scramjet.git
cd scramjet
npm ci
cd .d.ts
npx check-dts
```

This works and checks all the data, but when I change the node version to 15 (for instance `nvm i 15 && nvm use 15`).

```
/workspace/scramjet/.d.ts [develop|✚ 1] 
11:17 $ npx check-dts index.d.ts
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /workspace/scramjet/.d.ts/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/workspace/scramjet/.d.ts/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 
```

Then the check-dts checks seems to test files that should not be checked I guess... I get this error, no matter what the node version is:

```
npx check-dts
⠦ Check typesTypeError: Cannot read property 'text' of undefined
    at check (/workspace/scramjet/node_modules/check-dts/check.js:107:43)
    at async run (/workspace/scramjet/node_modules/check-dts/bin.js:32:18)
npm ERR! code 1
npm ERR! path /workspace/scramjet
npm ERR! command failed
npm ERR! command sh -c check-dts

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2020-10-27T11_56_46_567Z-debug.log
```

--- a/bin.js
+++ b/bin.js
@@ -5,7 +5,7 @@ let ciJobNumber = require('ci-job-number')
 
 if (ciJobNumber() !== 1) {
   process.stdout.write(
-    yellow('check-tds runs only on first CI job, to save CI resources\n')
+    yellow('check-dts runs only on first CI job, to save CI resources\n')
   )
   process.exit()
 }
@@ -32,9 +32,13 @@ async function run () {
     let result = await check(process.stdout, process.cwd(), print)
     if (!result) process.exit(1)
   } else {
-    error(`Unknown argument ${arg}\n`)
-    showHelp(print)
-    process.exit(1)
+    let result = await check(
+      process.stdout,
+      process.cwd(),
+      print,
+      process.argv.slice(2)
+    )
+    if (!result) process.exit(1)
   }
 }
 

--- a/check.js
+++ b/check.js
@@ -66,7 +66,12 @@ async function parseTest (files) {
   return expects
 }
 
-module.exports = async function check (stdout, cwd, print) {
+module.exports = async function check (
+  stdout,
+  cwd,
+  print,
+  globs = ['**/*.{js,ts,jsx,tsx}']
+) {
   let spinner = ora({ stream: stdout }).start('Check types')
 
   let compilerOptions
@@ -77,11 +82,17 @@ module.exports = async function check (stdout, cwd, print) {
   }
 
   let opts = { cwd, ignore: ['node_modules'], gitignore: true, absolute: true }
-  let all = await globby('**/*.{js,ts,jsx,tsx}', opts)
+  let all = (await Promise.all(globs.map(glob => globby(glob, opts)))).reduce(
+    (acc, x) => {
+      acc.push(...x)
+      return acc
+    },
+    []
+  )
 
   if (!all.some(i => /\.tsx?$/.test(i))) {
     let err = new Error(
-      'TypeScript files was not found. ' +
+      'TypeScript files were not found. ' +
         'Create .d.ts files and test/types.ts and test/errors.ts.'
     )
     err.own = true

--- a/show-help.js
+++ b/show-help.js
@@ -2,11 +2,14 @@ let { yellow: y, bold: b } = require('colorette')
 
 module.exports = function showHelp (print) {
   print(
-    b('Usage: ') + 'npx check-dts',
+    b('Usage: ') + 'npx check-dts [file1 file2...]',
     'Check `.d.ts` files in open source library according to types tests',
     '',
     b('Arguments:'),
+    '  ' +
+      y('file#') +
+      '       Optional a list of files/globs (otherwise matches all js/ts files)',
     '  ' + y('--version') + '   Show version',
-    '  ' + y('--help') + '   Show this message'
+    '  ' + y('--help') + '      Show this message'
   )
 }

